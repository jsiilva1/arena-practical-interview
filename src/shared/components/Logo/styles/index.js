import styled from 'styled-components';

import logoSrc from '../../../../../public/assets/images/logo-arena.svg';

import { device } from '../../../theme/settings/screens';

export const LogoContainer = styled.div`
  width: 100vw;
  display: flex;
  flex-flow: row wrap;
  margin: 0 auto;
  max-width: 465px;

  & p {
    margin: 2px 8px;
    font-weight: 400;
    color: #cdcdcd;
    font-size: 1.2rem;
  }

  @media ${device.tablet} {
    p {
      margin-top: 8px;
      font-size: 0.99rem;
    }
  }

  @media ${device.mobile} {
    float: left;
    margin: 0 10px;

    p { 
      font-size: 0.8rem; 
      margin-top: 5px;
    }
  }
`;

export const Figure = styled.figure`
  height: 23px;
  margin: 0;
`;

export const LogoAvatar = styled.img.attrs({
  src: logoSrc,
})`
  height: inherit;
`;
