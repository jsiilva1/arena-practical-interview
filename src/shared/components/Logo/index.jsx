/**
 * External Dependencies
 */
import React from 'react';

/**
 * Internal Dependencies
 */
import { LogoContainer, Figure, LogoAvatar } from './styles/index';

/**
 * Component Logo
 *
 * Render Logo of Arena Co
 * @return {*}
 */
const Logo = () => (
  <LogoContainer>
    <Figure>
      <LogoAvatar title="Arena.im" alt="Arena.im logo" />
    </Figure>

    <p>Practical Frontend Interview</p>
  </LogoContainer>
);

export default Logo;