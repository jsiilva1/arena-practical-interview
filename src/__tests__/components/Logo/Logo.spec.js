import React from 'react';
import { shallow } from 'enzyme';

import Logo from '../../../shared/components/Logo';

describe('Logo Component', () => {
  it('should render the Arena Company Logo correctly', () => {
    const wrapper = shallow(<Logo />);
    
    expect(wrapper).toMatchSnapshot();
  });
});
