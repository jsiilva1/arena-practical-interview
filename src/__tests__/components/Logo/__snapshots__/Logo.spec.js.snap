// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo Component should render the Arena Company Logo correctly 1`] = `
<styled.div>
  <styled.figure>
    <styled.img
      alt="Arena.im logo"
      title="Arena.im"
    />
  </styled.figure>
  <p>
    Practical Frontend Interview
  </p>
</styled.div>
`;
